kura/org.eclipse.kura.ai.triton.server/src/main/java/org/eclipse/kura/ai/triton/server/TritonServerServiceOptions.java

```diff
@@ -0,0 +1,117 @@
+package org.eclipse.kura.ai.triton.server;
+
+import static java.util.Objects.nonNull;
+import static java.util.Objects.requireNonNull;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class TritonServerServiceOptions {
+
+    private static final String PROPERTY_ADDRESS = "server.address";
+    private static final String PROPERTY_PORTS = "server.ports";
+    private static final String PROPERTY_LOCAL_MODEL_REPOSITORY_PATH = "local.model.repository.path";
+    private static final String PROPERTY_LOCAL_BACKENDS_PATH = "local.backends.path";
+    private static final String PROPERTY_LOCAL_BACKENDS_CONFIG = "local.backends.config";
+    private static final String PROPERTY_MODELS = "models";
+    private static final String PROPERTY_LOCAL = "enable.local";
+    private final Map<String, Object> properties;
+
+    private final int httpPort;
+    private final int grpcPort;
+    private final int metricsPort;
+    private final boolean isLocal;
+
+    public TritonServerServiceOptions(final Map<String, Object> properties) {
+        requireNonNull(properties, "Properties cannot be null");
+        this.properties = Collections.unmodifiableMap(new HashMap<>(properties));
+
+        final Object propertyPorts = this.properties.get(PROPERTY_PORTS);
+        if (nonNull(propertyPorts) && propertyPorts instanceof Integer[]) {
```

Comment:
If the `instanceof` expression returns true then `propertyPorts` is not null, there is no need for the `nonNull` check

Reply:
The same applies to the checks below